### PR TITLE
Use molecule-podman instead of molecule-docker

### LINKED ISSUE #36: Switch from molecule-docker to molecule-podman
In order to be prepared for the upcoming changes to the `ubuntu-latest` image in GitHub Actions, which is expected to run with `cgroupsv2´ enabled, we will be experiencing issues with systemd in docker.

Several posts seem to confirm, that Docker, systemd and `cgroupsv2` don't really work together:

* https://serverfault.com/a/1054414
* https://gist.github.com/pinkeen/bba0a6790fec96d6c8de84bd824ad933

One solution is to completely get rid of Docker and switch to Podman instead, which is what I suggest to do.

\cc @Normo and @christianhueserhzdr

### PR DESCRIPTION
**Pull request by tobiashuste**
This will help us to keep the tests running even on systems that have
`cgroupsv2` enabled. Docker is known to cause problems in combination
with systemd inside docker in this case

Closes #36

### CODE REVIEW

**Comment by tobiashuste** [comment]
@Normo and @christianhueserhzdr I'm mentioning you both as reviewer. This is a quite big change this chage would introduce to our roles. So I would be curious on your feedback.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,20 +53,26 @@ jobs:
...
+
       - name: Prepare the job environment.
         uses: ./.github/workflows/prepare-action
 
       - name: Run Molecule tests.
-        run: poetry run molecule test
+        run: XDG_RUNTIME_DIR=/run/user/$UID poetry run molecule test
         env:
-          MOLECULE_DISTRO: ${{ matrix.distro }}
+          MOLECULE_IMAGE: ${{ matrix.image }}
```
Comment (Normo) [inline .github/workflows/ci.yml:75]: I wonder why this works.. shouldn't that be `MOLECULE_IMAGE: ${{ matrix.distro }}` ?

Comment (tobiashuste) [reply .github/workflows/ci.yml:75]: Probably it always falls back to the default and never uses different images.

**Comment by tobiashuste** [comment]
@Normo This is how I would suggest to implement the test basing on Molecule. Could you please review?

**Comment by tobiashuste** [comment]
@christianhueserhzdr You force-pushed to this branch and thereby removed commit and changes? Is there any reason for this?

**Comment by christianhueserhzdr** [comment]
> @christianhueserhzdr You force-pushed to this branch and thereby removed commit and changes? Is there any reason for this?

Tobias @tobiashuste, I am very sorry. I did not realized that I removed something. I had to rebase the branch to add the changes from main and had to resolve conflicts. What did I remove?

**Comment by tobiashuste** [comment]
> > @christianhueserhzdr You force-pushed to this branch and thereby removed commit and changes? Is there any reason for this?
> 
> Tobias @tobiashuste, I am very sorry. I did not realized that I removed something. I had to rebase the branch to add the changes from main and had to resolve conflicts. What did I remove?

Thanks for the info. I will try to recover from my local version. Typically, I guess it's the better workflow, especially for more complex changes, if the author rebases themself.

**Comment by christianhueserhzdr** [comment]
> I was able to recover my changes tada

Phew, I am glad to hear that.

**Pull request merged**